### PR TITLE
Adds scaling for building footprint using building area

### PR DESCRIPTION
**Pull request by tanushree04**
### Pull Request Description

Adds optional argument to scale_footprint_area in the urban geometry creation measure, from https://github.com/NREL/openstudio-urban-measures/blob/transformers/measures/urban_geometry_creation/measure.rb. 

### Checklist (Delete lines that don't apply)

- [x] Unit tests have been added or updated
- [ ] Documentation has been modified appropriately
- [x] All ci tests pass (green)
- [x] An [ISSUE](https://github.com/urbanopt/urbanopt-geojson-gem/issues) has been created that this is addressing. Issues will get added to the Change Log when the change_log.rb script is run.
- [x] This branch is up-to-date with develop

### CODE REVIEW

File: lib/urbanopt/geojson/helper.rb [hunk 1]
```diff
@@ -207,9 +209,35 @@ def self.floor_print_from_polygon(polygon, elevation, origin_lat_lon, runner, zo
...
+          end
+        end
+
         return floor_print
       end
 
+      # scale footprint to desired area, keeping the shape
+      def self.adjust_vertices_to_area(vertices, desired_area, eps = 0.1, floor_print, runner)
+        ar = ScaleArea.new(vertices, desired_area, eps, floor_print, runner)
+        n = Newton::nlsolve(ar,[0])
```
Comment (tanushree04) [inline lib/urbanopt/geojson/helper.rb:236]: @kflemin I'm not sure what this line does

File: lib/urbanopt/geojson/scale_area.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        @runner = runner
+        @zero = BigDecimal::new("0.0")
+        @one  = BigDecimal::new("1.0")
+        @two  = BigDecimal::new("2.0")
+        @ten  = BigDecimal::new("10.0")
+        #@eps  = eps #BigDecimal::new(eps)
+      end
+
+      # compute value
+      def values(x)
```
Comment (tanushree04) [inline lib/urbanopt/geojson/scale_area.rb:65]: @kflemin I'm not sure how x is calculated here

File: lib/urbanopt/geojson/helper.rb [hunk 3]
```diff
@@ -207,9 +209,35 @@ def self.floor_print_from_polygon(polygon, elevation, origin_lat_lon, runner, zo
...
+
+        #check for scaling
+        if scaled_footprint_area > 0
+
+          # check that the scaled_footprint_area desired is no less than X % of the original
+          original_floor_print_area = OpenStudio::getArea(floor_print).get
+
+          if scaled_footprint_area / original_floor_print_area <= 0.5 || scaled_footprint_area / original_floor_print_area >= 2
+            # TOO MUCH SCALING...using original footprint when scaled is 2x bigger or smaller than the original
+            runner.registerWarning("Desired scaled_footprint_area is a factor of 2 of the original footprint...keeping original footprint (no scaling!)")
```
Comment (tanushree04) [inline lib/urbanopt/geojson/helper.rb:221]: @kflemin , the code is working correctly for this condition, but is not scaling to the new area in the else statement.

File: lib/urbanopt/geojson/scale_area.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        @new_vertices = URBANopt::GeoJSON::Zoning.divide_floor_print(@vertices, x[0].to_f, @runner, scale = true)
+        new_area = OpenStudio::getArea(@new_vertices)
+        fail "Cannot compute area for '#{@new_vertices}'" if new_area.empty?
+        new_area = new_area.get
+        
+        puts "x = #{x[0].to_f}, new_area = #{new_area}"
+        
+        return [new_area-@desired_area]
```
Comment (tanushree04) [inline lib/urbanopt/geojson/scale_area.rb:73]: Do we need this line? The `adjust_vertices_to_area` method only returns new_vertices.

File: lib/urbanopt/geojson/zoning.rb [hunk 5]
```diff
@@ -74,33 +75,38 @@ def self.divide_floor_print(floor_print, perimeter_depth, runner)
...
-            perim_vertices << vertices[i]
-            perim_vertices << vertices[0]
-            perim_vertices << new_vertices[0]
-            perim_vertices << new_vertices[i]
-          else
-            perim_vertices << vertices[i]
-            perim_vertices << vertices[i + 1]
-            perim_vertices << new_vertices[i + 1]
-            perim_vertices << new_vertices[i]
+        unless scale = true
```
Comment (kflemin) [inline lib/urbanopt/geojson/zoning.rb:78]: this should be == true.  Also how are you testing this to see that it doesn't go in the else clause? The tests are all passing so I'm not sure how to test it...

Comment (kflemin) [reply lib/urbanopt/geojson/zoning.rb:78]: when making this change, I get 4 test failures...maybe that will help narrow down the issue @tanushree04 ?

File: lib/urbanopt/geojson/scale_area.rb (hunk 2 above)
Comment (tanushree04) [reply lib/urbanopt/geojson/scale_area.rb:69]: It seems to be passing in 0 from `n = Newton::nlsolve(ar,[0])` . The area is not being scaled currently.

File: lib/urbanopt/geojson/zoning.rb (hunk 5 above)
Comment (tanushree04) [reply lib/urbanopt/geojson/zoning.rb]: @kflemin I updated to ==true and rewrote the logic its not failing for the else condition now

File: lib/urbanopt/geojson/scale_area.rb [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        @runner = runner
+        @zero = BigDecimal::new("0.0")
+        @one  = BigDecimal::new("1.0")
+        @two  = BigDecimal::new("2.0")
+        @ten  = BigDecimal::new("10.0")
+        @eps  = eps #BigDecimal::new(eps)
+      end
+
+      # compute value
+      def values(x)
```
Comment (tanushree04) [inline lib/urbanopt/geojson/scale_area.rb:68]: `n = Newton::nlsolve(ar,[0])` is iterating over different values for perimeter depth x. It is failing for 1 with this error  - `undefined method `zero' for #<URBANopt::GeoJSON::ScaleArea:0x0000000708efd8>`

**Pull request merged**